content/blog/using-mcps-to-run-terraform.md

```diff
@@ -0,0 +1,216 @@
+
+
+Example Cursor `mcp.json` config file
+```json
+{
+  "mcpServers": {
+    "tfmcp": {
+      "command": "/Users/weston/clients/westonplatter/tfmcp/target/release/tfmcp",
+      "args": [
+        "mcp"
+      ],
+      "env": {
+        "HOME": "/Users/weston",
+        "PATH": "/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/homebrew/bin",
+        "TERRAFORM_DIR": "/Users/weston/clients/westonplatter/example-devops"
+      }
+    }
+  }
+} 
+```
+
+
+### Terraform code for us to run
+From there, I added a `main.tf` file in the configured `TERRAFORM_DIR` and wrote some terraform code:
+* **Terraform block**: sets the required provider, `cyrilgdn/postgresql`
+* **Provider config**: connects to a local Postgres instance as `admin_user`
+* **Postgres Role resource**: creates a login-enabled role for the app
+* **Postgres Database resource**: creates a DB owned by the app role
+* **Schema resource**: creates a schema in the DB owned by the app role
+
+```terraform
+# declare terraform providers
+terraform {
+ required_providers {  
+   postgresql  = {  
+     source    = "cyrilgdn/postgresql"
+     version   = "1.25.0"
+   }  
+ }  
+}
+
+# configure the postgres provider with auth credentials for the postgres db
+provider "postgresql" {  
+ host             = "localhost"
+ port             = 5432  
+ database         = "postgres"  
+ username         = "admin_user"  
+ password         = ""  
+ sslmode          = "disable"  
+ connect_timeout  = 15  
+}
+
+# create a role for the app  
+resource "postgresql_role" "app_role" {  
+ name     = "app_role"
+ login    = true  
+ password = "mypass" # intentionally insecure password for demo
+}
+
+# create a database for the app
+resource "postgresql_database" "app_db" {  
+  name                   = "app_db"  
+  owner                  = postgresql_role.app_role.name  
+  template               = "template0"  
+  lc_collate             = "C"  
+  connection_limit       = -1  
+  allow_connections      = true  
+  alter_object_ownership = true  
+}
+
+# create a schema for the app  
+resource "postgresql_schema" "app_schema" {  
+  name     = "app_schema"  
+  owner    = postgresql_role.app_role.name  
+  database = postgresql_database.app_db.name  
+}
+```
+
+<br>
+
+### Debugging terraform apply
+At this point, I was ready for the agent to run terraform plan and apply my changes. And here is where the rubber meets the road in using MCPs 🙂. Here’s what I ran into:
+
+**The first issue** was a permissions error — `admin_user` couldn’t create Postgres roles. This issue was totally my fault — I hadn’t granted it any meaningful privileges. I had the MCP repeatedly run `terraform apply`, but either the MCP didn't log/surface the error or Cursor’s Agent Mode wasn't able to interpret the cause of the error.
+![First issue - MCP able to run terraform apply](/img/updates/using-mcps-to-run-terraform/debug1.png)
```

Comment:
@westonplatter this image is good, but it's hard to read even on a large screen and we sadly don't have a lightbox that'll show it in better resolution when you click on it. Can you please break this up into two screenshots so we have both bit bigger? They can be one on top of the other -- I just want to make sure people can read them.